src/Module/EmailTagRegisterHandler.php

```diff
@@ -0,0 +1,131 @@
+use Dhii\Util\Normalization\NormalizeStringCapableTrait;
+use Psr\Container\ContainerInterface;
+use RebelCode\Modular\Events\EventsFunctionalityTrait;
+use stdClass;
+use Traversable;
+
+/**
+ * Handler that registers the EDD email tags and their handlers with EDD.
+ *
+ * @since [*next-version*]
+ */
+class EmailTagRegisterHandler implements InvocableInterface
+{
+    /* @since [*next-version*] */
+    use ContainerAwareTrait;
+
+    /* @since [*next-version*] */
+    use ContainerGetCapableTrait;
+
+    /* @since [*next-version*] */
+    use NormalizeKeyCapableTrait;
+
+    /* @since [*next-version*] */
+    use NormalizeStringCapableTrait;
+
+    /* @since [*next-version*] */
+    use NormalizeIterableCapableTrait;
+
+    /* @since [*next-version*] */
+    use EventsFunctionalityTrait;
+
+    /* @since [*next-version*] */
+    use CreateContainerExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateNotFoundExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateInvalidArgumentExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateOutOfRangeExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateInternalExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateRuntimeExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use StringTranslatingTrait;
+
+    /**
+     * The config for the email tags.
+     *
+     * @since [*next-version*]
+     *
+     * @var array|stdClass|Traversable
+     */
+    protected $tagsConfig;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param array|stdClass|Traversable $tagsConfig The list of configs for the email tags to register.
+     * @param ContainerInterface         $container  The services container, for resolving tag handlers.
+     */
+    public function __construct($tagsConfig, ContainerInterface $container)
+    {
+        $this->_setTagsConfig($tagsConfig);
+        $this->_setContainer($container);
+    }
+
+    /**
+     * Retrieves the config for the email tags.
+     *
+     * @since [*next-version*]
+     *
+     * @return array|stdClass|Traversable
+     */
+    protected function _getTagsConfig()
+    {
+        return $this->tagsConfig;
+    }
+
+    /**
+     * Sets the config for the email tags.
+     *
+     * @since [*next-version*]
+     *
+     * @param array|stdClass|Traversable $tagsConfig The config for the email tags.
+     */
+    protected function _setTagsConfig($tagsConfig)
+    {
+        $this->tagsConfig = $this->_normalizeIterable($tagsConfig);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function __invoke()
+    {
+        foreach ($this->_getTagsConfig() as $_tag) {
+            \edd_add_email_tag(
+                $this->_containerGet($_tag, 'tag'),
+                $this->_translate($this->_containerGet($_tag, 'description')),
```

Comment:
The `__()` method should be used for i18n.

Reply:
But here I'm doing l10n.

Reply:
Internationalization is the process of handling strings such that they can be substituted by their translation. Localization is the process of providing those translations. How is this l10n?

Reply:
Because the translated string is required here to be passed. The string is obtained from elsewhere, and that's what should i18n it.

You'll notice that I'm obtaining the string via

```
$this->_containerGet($_tag, 'description')
```

This is not something that can be i18n - not by _not_ running the code anyway.

Reply:
It doesn't matter what it is i18n-ized by. The i18n method is `__()`. The `_translate()` method is also an i18n method (not l10n, because there can't be a method for that), but `_translate()` is not standards-compliant.